src/write.c

```diff
@@ -235,11 +235,18 @@ typedef struct avifEncoderData
 
 static void avifEncoderDataDestroy(avifEncoderData * data);
 
+// Returns NULL if a memory allocation failed.
 static avifEncoderData * avifEncoderDataCreate()
 {
     avifEncoderData * data = (avifEncoderData *)avifAlloc(sizeof(avifEncoderData));
+    if (!data) {
+        return NULL;
+    }
     memset(data, 0, sizeof(avifEncoderData));
     data->imageMetadata = avifImageCreateEmpty();
+    if (!data->imageMetadata) {
+        goto error;
+    }
```

Comment:
In `avifEncoderDataDestroy()`, we need to handle a null `data->imageMetadata` now:

```
static void avifEncoderDataDestroy(avifEncoderData * data)
{
    ...
    if (data->imageMetadata) {
        avifImageDestroy(data->imageMetadata);
    }
    ...
}
```